packages/nimble-components/src/utilities/tests/parameterized.ts

```diff
@@ -64,3 +30,117 @@ export const getSpecTypeByNamedList = <T extends { name: string }>(
     (x: T) => focusList.includes(x.name),
     (x: T) => disabledList.includes(x.name)
 );
+
+// eslint-disable-next-line no-restricted-globals
+type Fit = typeof fit;
+type Xit = typeof xit;
+type It = typeof it;
+/**
+ * One of the jasmine spec functions: fit, xit, or it
+ */
+type Spec = Fit | Xit | It;
+/**
+ * One of the jasmine spec functions: fit or xit
+ */
+type SpecOverride = Fit | Xit;
+
+/**
+ * Used to create a parameterized test using an object of test names and arbitrary test values.
+ * In the following example:
+ *  - the test named `catsAndDogs` is focused for debugging
+ *  - the test named `frogs` is configured to always be disabled
+ *  - the test named `men` will run normally as it has no override
+ * @example
+ * const rainTests = {
+ *     catsAndDogs: 'idiom',
```

Comment:
nit: is it possible to use a syntax that allows for spaces in the names of the tests? That would make the specs read a bit easier.

Not critical if it requires a major refactor but if it's easy to just change these examples to quote the property names then I'd vote to encourage readability in the printed test name.

Reply:
Yes... it is possible, but wondering what our guidance should be.

Example 1:

```ts
const rainTests = {
    // eslint-disable-next-line @typescript-eslint/naming-convention
    'really really really really really long test name cats and dogs': {
        testData: 0
    },
} as const;
describe('Different rains', () => {
    parameterize(rainTests, (spec, name, value) => {
        spec(`of type ${name} exist`, () => {
            expect(value).toBeDefined();
        });
    }, {
        // eslint-disable-next-line @typescript-eslint/naming-convention
        'really really really really really long test name cats and dogs': fit,
    });
});
```
Pros:
- Don't need to think of an identifier for each test case. These names get [really long](https://github.com/ni/nimble/blob/931138bb9d3601051afc3e394885a8a90e9dc83b/packages/nimble-components/src/table/tests/table-column-sizing.spec.ts#L185) and nuanced so I guess it's nice to not need to think of a short identifier name that represents this case.

Cons:
- We would need to update our eslint rules to allow for funky key names. Maybe we just do it for tests? Or do we have eslints on each line? Maybe there is a rule config we can loosen vs completely disable in tests
- Makes the overrides look clunky but that's not a big deal really

Example 2:
```ts
const rainTests = {
    catsAndDogs: {
        specName: 'really really really really really long test name cats and dogs',
        testData: 0
    },
} as const;
describe('Different rains', () => {
    parameterize(rainTests, (spec, _name, value) => {
        spec(`of type ${value.specName} exist`, () => {
            expect(value).toBeDefined();
        });
    }, {
        catsAndDogs: fit
    });
});
```

Pros:
- More normal configuration object shape
- More normal keys for disabling in override
- Semi forced / encouraged to come up with a shorter name, could maybe help with distinguishing a test case

Cons:
- A fair bit more mental energy to think of test case names
- End up with a frequently unused `name` parameter

Reply:
Consensus with coding practices WGwas to start with Example 1 and consider changing our eslint defaults so that naming-convention doesn't fire for this case

Reply:
Made a pattern that allows configuring via the named list pattern (which also aligns more closely with that already exists) and I think it avoids all the trade-offs discussed.